clang/lib/Headers/__clang_cuda_intrinsics.h

```diff
@@ -479,6 +479,275 @@ inline __device__ unsigned __funnelshift_rc(unsigned low32, unsigned high32,
   return ret;
 }
 
+#define INTRINSIC_LOAD(func_name, asm_op, decl_type, internal_type, asm_type) \
```

Comment:
Can we merge `INTRINSIC*` and `MINTRINSIC*` macros by providing an additional clobber parameter, which could contain either nothing  (e.g `MACRO(something, )` ) or `: "memory"` (`MACRO(something, : "memory" )`). It may look a bit odd, but it appears to work: https://godbolt.org/z/98vqn67qT